TorrentCore/Tracker/Http/HttpTracker.cs

```diff
@@ -64,23 +64,16 @@ public virtual async Task<AnnounceResult> Announce(AnnounceRequest request)
 
         private async Task<Stream> SendRequest(AnnounceRequest request)
         {
-            byte[] peerId = new byte[20];
-            Array.Copy(Encoding.UTF8.GetBytes("-AZ5501-"), peerId, 8);
-            byte[] rand = new byte[12];
-            Random r = new Random();
-            r.NextBytes(rand);
-            Array.Copy(rand, 0, peerId, 8, 12);
-
             // Prepare query
             StringBuilder queryBuilder = new StringBuilder();
             queryBuilder.Append("?event=started");
             if (_tcpConnectionDetails.PublicAddress != null)
                 queryBuilder.Append(string.Format("&ip={0}", _tcpConnectionDetails.PublicAddress));
             queryBuilder.Append(string.Format("&port={0}", _tcpConnectionDetails.Port)); // TODO: use public port
-            queryBuilder.Append(string.Format("&peer_id={0}", Encoding.UTF8.GetString(WebUtility.UrlEncodeToBytes(peerId, 0, peerId.Length))));
+            queryBuilder.Append(string.Format("&peer_id={0}", request.PeerId));
```

Comment:
This will still need to be URL-encoded. That could either be done here, or in the `ToString()` method of `PeerId` because it currently returns a string that is not safe to include directly in the URL.